x/trigger/simulation/genesis.go

```diff
@@ -0,0 +1,140 @@
+package simulation
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"time"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	sdktx "github.com/cosmos/cosmos-sdk/types/tx"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+
+	"github.com/provenance-io/provenance/internal/pioconfig"
+	"github.com/provenance-io/provenance/x/trigger/types"
+)
+
+const (
+	TriggerID  = "trigger_id"
+	QueueStart = "queue_start"
+)
+
+// TriggerIDFn randomized MinActions
+func TriggerIDFn(r *rand.Rand) uint64 {
+	return uint64(simtypes.RandIntBetween(r, 2, 10000000000))
+}
+
+// QueueStartFn randomized MinActions
+func QueueStartFn(r *rand.Rand) uint64 {
+	return uint64(simtypes.RandIntBetween(r, 0, 10000000000))
+}
+
+// GetRandomEvent returns a random event
+func GetRandomEvent(r *rand.Rand, now time.Time) types.TriggerEventI {
+	rand := simtypes.RandIntBetween(r, 0, 1)
+	if rand > 0 {
+		randTime := now.Add(time.Hour * time.Duration(simtypes.RandIntBetween(r, 1, 10)))
```

Comment:
I've never tried to see what happens to the blocktime during simulations. If they don't do anything special, then this should be shortened to something from maybe 10 seconds to 5 minutes. There's a chance that they manually alter the block time, though, to make the simulation think the blocks are taking longer than they actually are.

The goal is to pick a duration here that has a good chance of being encountered while the simulation runs. It's okay if some of them are so far out the simulation won't get to them, but we definitely want some to be triggered.